docs/src/main/sphinx/connector/hudi.md

```diff
@@ -82,6 +82,11 @@ Additionally, following configuration properties can be set depending on the use
   - Maximum number of metastore data objects per transaction in the Hive
     metastore cache.
   - `2000`
+* - `hudi.query-partition-filter-required`
+  - Set to `true` to force a query to use a partition filter. You can use the
+    `query_partition_filter_required` catalog session property for temporary,
+    catalog specific use.
+  - `false`
```

Comment:
Thanks for the contribution @krvikash . I am yet to review the full code. But, just for my understanding, why should the default be false? Is the plan produced an unoptimized one if the query uses a partition filter? I would be helpful if you can paste the plans for a simple query with partition filter with and without this config.

Reply:
Hi @codope, This PR aims to mandate the inclusion of partition filtering in SELECT queries. This prevents accidental execution of `SELECT *` queries on tables containing substantial amount of data. Notably, this enforcement won't alter the query plan.

Reply:
Thanks for the clarification. I am still not sure why the default is false if it's a good thing?

Reply:
We prefer not to enforce this unless explicitly instructed to do so, following the approach of Hive, Delta, and Iceberg.

Reply:
Ok, I am trying to understand the reasoning. As a user, I think it makes more sense to enable this by default. Do you know why it is false for other connectors?

Reply:
I don't know why it is false for other connectors.

But enabling the `delta.query-partition-filter-required` property by default will fail some queries that do not meet the partition column requirement. It will be unexpected for the existing user.

Reply:
>  Do you know why it is false for other connectors?

@Praveen2112 do you have any idea on this?

Reply:
> But enabling the delta.query-partition-filter-required property by default will fail some queries that do not meet the partition column requirement. It will be unexpected for the existing user.

Exactly - This config avoid hitting potentially long running queries (for larger tables) but enabling it default might restrict the set of queries which would be executed on Trino for smaller tables as well.

Reply:
@codope does this answer your question?

Reply:
Yes thanks @krvikash. I think we should document the full behavior, if possible with an example. User should be aware of side-effects of enabling this config.

Reply:
@codope Thanks for review. Added the behavior in the doc. Please take a look.

Reply:
the other part is that depending on the expressions used in the predicate we may or may not be able to understand that the partitions are actually filtered leading to false positives too.